server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java

```diff
@@ -0,0 +1,262 @@
+            "Memory, in bytes, used by the young generation heap.",
+            "bytes",
+            () -> new LongWithAttributes(bytesUsedByGCGen(stats.getOrRefresh().getJvm().getMem(), GcNames.YOUNG))
+        );
+
+        registry.registerLongGauge(
+            "es.node.stats.jvm.mem.pools.survivor.used",
+            "Memory, in bytes, used by the survivor space.",
+            "bytes",
+            () -> new LongWithAttributes(bytesUsedByGCGen(stats.getOrRefresh().getJvm().getMem(), GcNames.SURVIVOR))
+        );
+
+        registry.registerLongGauge(
+            "es.node.stats.jvm.mem.pools.old.used",
+            "Memory, in bytes, used by the old generation heap.",
+            "bytes",
+            () -> new LongWithAttributes(bytesUsedByGCGen(stats.getOrRefresh().getJvm().getMem(), GcNames.OLD))
+        );
+
+        registry.registerLongAsyncCounter(
+            "es.node.stats.fs.io_stats.io_time.total",
+            "The total time in millis spent performing I/O operations across all devices used by Elasticsearch.",
+            "milliseconds",
+            () -> new LongWithAttributes(stats.getOrRefresh().getFs().getIoStats().getTotalIOTimeMillis())
+        );
+    }
+
+    /**
+     * Retrieves the bytes used by a specific garbage collection generation from the provided JvmStats.Mem.
+     *
+     * @param mem  The JvmStats.Mem containing memory pool information.
+     * @param name The name of the garbage collection generation (e.g., "young", "survivor", "old").
+     * @return The number of bytes used by the specified garbage collection generation.
+     */
+    private long bytesUsedByGCGen(JvmStats.Mem mem, String name) {
+        long bytesUsed = 0;
+        for (JvmStats.MemoryPool pool : mem) {
+            if (pool.getName().equals(name)) {
+                bytesUsed = pool.getUsed().getBytes();
+            }
+        }
+        return bytesUsed;
+    }
+
+    /**
+     * Retrieves the current NodeStats for the Elasticsearch node.
+     *
+     * @return The current NodeStats.
+     */
+    private NodeStats getNodeStats() {
+        CommonStatsFlags flags = new CommonStatsFlags(
+            CommonStatsFlags.Flag.Get,
+            CommonStatsFlags.Flag.Search,
+            CommonStatsFlags.Flag.Merge,
+            CommonStatsFlags.Flag.Translog
+        );
+        return nodeService.stats(
+            flags,
+            true,
+            false,
+            false,
+            true,
+            false,
+            true,
+            true,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false
+        );
+    }
+
+    @Override
+    protected void doStart() {
+        registerAsyncMetrics(registry);
+    }
+
+    @Override
+    protected void doStop() {
+        stats.stopRefreshing();
+    }
+
+    @Override
+    protected void doClose() throws IOException {}
```

Comment:
the try-with-resources is not usable here, since the `AutoClosable` is the metrics itself, therefore we would close the metric right after it has being registered:
```
try(registry.registerLongGauge(...)){
    //do nothing
} 
// here the close() method would have being called therefore de-registering the metric
```
Instead I have an ArrayList of all the metrics and call `close()` on all counters and gauges in the `doClose()` of the `AbstractLifecycleComponent`.
And being doClose called once is safe to de-register the metrics there.
Java doc for the doClose method:
> It is called once in the lifetime of a component. If the component was started then it will be stopped before it is closed, and once it is closed it will not be started or stopped.